controllers/idler/idler_controller.go

```diff
@@ -109,8 +114,8 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 	}
 	// Find the earlier pod to kill and requeue. Otherwise, use the idler timeoutSeconds to requeue.
 	nextTime := nextPodToBeKilledAfter(logger, idler)
-	if nextTime == nil {
-		after := time.Duration(idler.Spec.TimeoutSeconds) * time.Second
+	after := time.Duration(RequeueTimeThresholdSeconds) * time.Second
+	if nextTime == nil || (nextTime != nil && *nextTime > after) {
```

Comment:
```suggestion
	if nextTime == nil || *nextTime > after {
```
You don't really need to check nextTime != nil in the second condition because if it's nil it won't get there. If the first condition is true (nextTie == nil) then it does't execute the second one (this is how `||` works).

Reply:
I had the same understanding. I wasn't checking for `!= nil` initially, but later fixed it in [d40b75b](https://github.com/codeready-toolchain/member-operator/pull/630/commits/d40b75bd609843b384e6dd0b6397ca9cda4f15ca) because I was getting segmentation  fault error when i was testing it in go playground. I don't have have the previous playground to look up what i wrote but I guess I was just doing something wrong before - https://go.dev/play/p/pPIV6OuIeDG
reverting the condition now. Thanks!

Reply:
reverted in [c230065](https://github.com/codeready-toolchain/member-operator/pull/630/commits/c2300653882c00715be041bc59c395a5db195305)